packages/KDropdownMenu/KDropdownMenu.vue

```diff
@@ -0,0 +1,239 @@
+<template>
+  <div
+    :class="{'selection-dropdown-menu': appearance === 'selectionMenu'}"
+    class="k-dropdown k-dropdown-menu"
+  >
+    <KToggle v-slot="{toggle, isToggled}">
+      <KPop
+        v-bind="boundKPopAttributes"
+        :on-popover-click="() => {
+          toggle();
+          return isToggled
+        }"
+        @opened="() => {
+          toggle()
+          $emit('toggleDropdown', true)
+        }"
+        @closed="() => {
+          toggle()
+          $emit('toggleDropdown', false)
+        }"
+      >
+        <div class="d-flex">
+          <component
+            :is="!!disabledTooltip ? 'Kooltip' : 'div'"
+            :label="disabledTooltip"
+            :position="!!disabledTooltip ? 'bottom' : undefined"
+            :position-fixed="!!disabledTooltip ? true : undefined"
+            :max-width="!!disabledTooltip ? '240' : undefined"
+            class="dropdown-trigger"
+          >
+            <slot
+              :is-open="isToggled"
+              name="default"
+            >
+              <!-- Must wrap in div to allow tooltip when disabled -->
+              <div>
+                <KButton
+                  v-if="label"
+                  :disabled="disabled"
+                  :is-open="showCaret || appearance === 'selectionMenu' ? isToggled : undefined"
+                  :class="{ 'is-active': showCaret ? isToggled : undefined }"
+                  :appearance="appearance === 'selectionMenu' ? 'outline' : 'primary'"
+                  class="k-dropdown-btn"
+                >
+                  {{ label }}
+                </KButton>
+              </div>
+            </slot>
+          </component>
+        </div>
+        <template #content>
+          <ul class="k-dropdown-list dropdown-list">
+            <slot
+              :items="items"
+              :is-open="isToggled"
+              name="items"
+            >
+              <KDropdownItem
+                v-for="(item, idx) in items"
+                v-bind="item"
+                :key="`${item.label.replace(' ', '-')}-${idx}`"
```

Comment:
As long as the label is a string, you shouldn't need to strip characters. If you are going to strip them for the key, I would strip other "typical" invalid characters like quotes, etc. that normally wouldn't be allowed in an HTML attribute.

Reply:
If spaces won't break it I'll just remove the `replace()`

Reply:
Done